### PR TITLE
handle malformed urls

### PR DESCRIPTION
**Pull request by FlorianCassayre**
* addresses inveniosoftware/react-invenio-app-ils#128
* also exposes the class `UrlParamValidator` in case one needs to build a stricter validator on top of it (for instance, ensure the provided page size is indeed an option of the dropdown)

### CODE REVIEW

File: src/lib/api/UrlParamValidator.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      case 'sortOrder':
+        return ['asc', 'desc'].includes(value);
+      case 'page': case 'size':
+        return _isNumber(value) && value > 0;
+      case 'layout':
+        return ['grid', 'list'].includes(value);
+      case 'filters': case 'hiddenParams':
+        const array = _isArray(value) ? value : [value];
+        const cKvSep = ':', cChildSep = urlHandler.urlFilterSeparator;
+        const pLiteral = `[^\\${cKvSep}\\${cChildSep}]*`,
```
Comment (ntarocco) [inline src/lib/api/UrlParamValidator.js:33]: oh wow, regex pro here! :D 👏

File: src/lib/api/UrlParamValidator.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+export class UrlParamValidator {
+  /**
+   * Return true if the param value is valid, false otherwise
+   * @param {object} urlHandler the url handler
+   * @param {string} key the parameter key
+   * @param {string} value the parameter value
+   */
+  isValid = (urlHandler, key, value) => {
+    switch (key) {
+      case 'queryString': case 'sortBy':
```
Comment (ntarocco) [inline src/lib/api/UrlParamValidator.js:22]: funny formatting... I couldn't see the `sortBy`!

**Review by ntarocco — APPROVED** [review]
Just to understand, this is to validate the value of each param?
When we created, I think we thought that if values are wrong, it is because of URL manipulation and then it is out of RSK context to handle this.
But since you did, good job! :)

**Comment by FlorianCassayre** [comment]
I see, I was not 100% sure about this part but it can be changed for sure (raise an exception instead of filtering the invalid parameters). Or even better, have it as a configurable behaviour. One argument in favour of filtering was to be able to gracefully handle old urls after one of the parameter name has changed; or for instance the allowed page sizes are updated. 
In any case the existing validation was incomplete and would let invalid values pass through, which we don't want for sure.

File: src/lib/api/UrlParamValidator.js (hunk 1 above)
Comment (FlorianCassayre) [reply src/lib/api/UrlParamValidator.js:33]: (a bit verbose but I can make i a for-loop if too confusing)

File: src/lib/api/UrlParamValidator.js (hunk 2 above)
Comment (FlorianCassayre) [reply src/lib/api/UrlParamValidator.js:22]: Thanks, will add a line-break.

**Comment by FlorianCassayre** [comment]
Post-discussion: yes filter them, no need for config.

**Review by ntarocco — APPROVED** [review]

**Pull request merged**